php/datasource/class-fieldmanager-datasource-user.php

```diff
@@ -114,146 +137,197 @@ public function get_value( $value ) {
+	 */
+	public function get_ajax_action() {
+		if ( ! empty( $this->ajax_action ) ) {
+			return $this->ajax_action;
+		}
+		$unique_key = wp_json_encode( $this->query_args );
+		$unique_key .= $this->display_property;
+		$unique_key .= (string) $this->query_callback;
+		return 'fm_datasource_post' . crc32( $unique_key );
+	}
+
+	/**
+	 * Delete reciprocal user metadata prior to saving (presave will re-add).
+	 * Reciprocal relationships are not possible if we are not storing by ID.
+	 *
+	 * @param Fieldmanager_Field $field          The current field.
+	 * @param array              $values         The new values.
+	 * @param array              $current_values The existing values.
+	 * @return string                            Sanitized values.
+	 */
+	public function presave_alter_values( Fieldmanager_Field $field, $values, $current_values ) {
+		if ( 'post' != $field->data_type || ! $this->reciprocal || 'ID' != $this->store_property ) {
 			return $values;
 		}
 
 		if ( ! empty( $current_values ) ) {
 			foreach ( $current_values as $user_id ) {
-				delete_user_meta( $user_id, $this->reciprocal, $field->data_id );
+				call_user_func(
+					/**
+					 * Allow control over deleting from the user data storage.
+					 * This improves compatibility with WordPress.com.
+					 *
+					 * @see delete_user_meta() for more details about each param.
+					 *
+					 * @param string $function_name The function to call to get user
+					 *                              data. Default is 'delete_user_meta'.
+					 * @param int $user_id User ID.
+					 * @param string $meta_key Meta key to retrieve.
+					 * @param mixed $meta_value Only delete if the current value matches.
+					 */
+					apply_filters( 'fm_user_context_delete_data', 'delete_user_meta' ),
+					$user_id,
+					$this->reciprocal,
+					$field->data_id
+				);
+			}
+		}
+
+		return $values;
+	}
+
+	/**
+	 * Handle reciprocal usermeta.
+	 * Reciprocal relationships are not possible if we are not storing by ID.
+	 *
+	 * @param  Fieldmanager_Field $field         The current field.
+	 * @param  array              $value         The new value.
+	 * @param  array              $current_value The existing value.
+	 * @return string                            Sanitized value.
+	 */
+	public function presave( Fieldmanager_Field $field, $value, $current_value ) {
+		if ( empty( $value ) ) {
+			return;
+		}
+
+		$return_single = false;
+		if ( ! is_array( $value ) ) {
+			$return_single = true;
+			$value = array( $value );
+		}
+
+		foreach ( $value as $i => $v ) {
+			$value[ $i ] = $this->sanitize_value( $v );
+			if ( ! current_user_can( $this->capability, $v ) ) {
+				/* translators: %s: user id */
+				wp_die( esc_html( sprintf( __( 'Tried to refer to user "%s" which current user cannot edit.', 'fieldmanager' ), $v ) ) );
+			}
+			if ( $this->reciprocal && 'ID' == $this->store_property ) {
+				call_user_func(
+					/**
+					 * Allow control over adding to the user data storage.
```

Comment:
Per the docs standards, [filter summaries should begin with "Filters..."](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#language).